panda/src/physics/linearSinkForce.cxx

```diff
@@ -61,7 +61,25 @@ make_copy() {
  */
 LVector3 LinearSinkForce::
 get_child_vector(const PhysicsObject *po) {
-  return (get_force_center() - po->get_position()) * get_scalar_term();
+  LVector3 distance_vector = get_force_center() - po->get_position();
+  PN_stdfloat distance = distance_vector.length();
```

Comment:
Let's do `length_squared()` here, that is more efficient since it doesn't do the `sqrt`. Later you can do a `sqrt` in the two cases where it is necessary only.